ext/spl/spl_observer.stub.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+interface SplObserver
+{
+    /** @return void */
+    public function update(SplSubject $subject);
+}
+
+interface SplSubject
+{
+    /** @return void */
+    public function attach(SplObserver $observer);
+
+    /** @return void */
+    public function detach(SplObserver $observer);
+
+    /** @return void */
+    public function notify();
+}
+
+class SplObjectStorage implements Countable, Iterator, Serializable, ArrayAccess
+{
+    /**
+     * @param mixed $info
+     * @return void
+     */
+    public function attach(object $object, $info = null) {}
+
+    /** @return void */
+    public function detach(object $object) {}
+
+    /** @return bool */
+    public function contains(object $object) {}
+
+    /** @return int */
+    public function addAll(SplObjectStorage $storage) {}
+
+    /** @return void */
+    public function removeAll(SplObjectStorage $storage) {}
+
+    /** @return void */
+    public function removeAllExcept(SplObjectStorage $storage) {}
+
+    /** @return mixed */
+    public function getInfo() {}
+
+    /**
+     * @param mixed $info
+     * @return void
+     */
+    public function setInfo($info) {}
+
+    /** @return int */
+    public function count(int $mode = COUNT_NORMAL) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return int */
+    public function key() {}
+
+    /** @return object */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return void */
+    public function unserialize(string $serialized) {}
+
+    /** @return string|null */
+    public function serialize() {}
+
+    /**
+     * @param object $object
+     * @return bool
+     */
+    public function offsetExists($object) {}
+
+    /**
+     * @param object $object
+     * @return string
+     */
+    public function offsetGet($object) {}
+
+    /**
+     * @param object $object
+     * @param mixed $info
+     * @return void
+     */
+    public function offsetSet($object, $info = null) {}
+
+    /**
+     * @param object $object
+     * @return void
+     */
+    public function offsetUnset($object) {}
+
+    /** @return string */
+    public function getHash(object $object) {}
+
+    /** @return array */
+    public function __serialize() {}
+
+    /** @return void */
+    public function __unserialize(array $data) {}
+}
+
+class MultipleIterator implements Iterator
+{
+    public function __construct(int $flags = MultipleIterator::MIT_NEED_ALL|MultipleIterator::MIT_KEYS_NUMERIC) {}
+
+    /** @return int */
+    public function getFlags() {}
+
+    /** @return void */
+    public function setFlags(int $flags) {}
+
+    /**
+    * @param Iterator $iterator
+    * @param int|string|null $info
+    * @return void
+    */
+    public function attachIterator(Iterator $iterator, $info = null) {}
+
+    /** @return void */
+    public function detachIterator(object $iterator) {}
```

Comment:
Hm... one of the rare cases where we have pre-existing type hints, which say `Iterator $Iterator` here. However, the implementation is shared with SplObjectStorage which only checks `object`. This is an error introduced when I originally disabled arginfo checks for internal functions.

I think the proper way to do this would be to actually duplicate the three functions this affects, with adjusted zpp type checks.

Reply:
You mean `detachIterator()`, `containsIterator()`, and `countIterators()`, right? The first two is no problem, however as far as I can see, `countIterators()`  could be shared between the two classes. Or do I miss something?

Reply:
Yeah, off-by-one ;) Only `detachIterator()` and `containsIterator()` are problematic.